ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/extensions/IntSizeExtensions.kt

```diff
@@ -1,6 +1,26 @@
 package com.revenuecat.purchases.ui.revenuecatui.extensions
 
+import android.annotation.SuppressLint
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.onGloballyPositioned
 import androidx.compose.ui.unit.IntSize
+import com.revenuecat.purchases.ui.revenuecatui.data.PaywallState
+import com.revenuecat.purchases.ui.revenuecatui.data.isInFullScreenMode
 
 internal val IntSize.aspectRatio: Float
     get() = width.toFloat() / height
+
+internal val IntSize.isLandscape: Boolean
+    get() = width > height
+
+/**
+ * @return: Modifier with a callback indicating whether landscape layout should be used.
+ */
+@SuppressLint("ModifierFactoryExtensionFunction")
+internal fun PaywallState.Loaded.onLandscapeLayoutChanged(
+    changed: (Boolean) -> Unit,
+) = Modifier
+    .then(
+        // TODO: vertical compact?
```

Comment:
@tonidero I'd love your feedback on this.

On iOS we don't use the aspect ratio to determine landscape layout, but instead only use it if the "vertical size is compact" (which is basically only true for phones + landscape, but not landscape tablets).

Should we have some threshold screen height to make this false so landscape tablets still use the portrait layout?

Reply:
Hmm in Android, we have [window size classes](https://github.com/RevenueCat/purchases-android/blob/79e9c6bc6e9536dab13296723a603d28da186692/ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/helpers/WindowHelper.kt#L11). We do use that to calculate whether to display as a full screen dialog or not when presenting the paywall as a dialog. I think we might be able to use something like that?

In that example, I'm using the width size class, but we should also be able to calculate the height size class. As for the accuracy... It's provided by Google, so it should work, but we should test on different devices just in case.